src/mappings/colony.ts

```diff
@@ -118,12 +119,24 @@ export function handlePaymentAdded(event: PaymentAdded): void {
 
 export function handleColonyMetadata(event: ColonyMetadata): void {
   let colony = Colony.load(event.address.toHexString())
-  colony.metadata = event.params.metadata.toString()
-  // NB you have to do this reassignment dance with arrays
-  let metadataHistory = colony.metadataHistory
-  metadataHistory.push(event.params.metadata.toString())
-  colony.metadataHistory = metadataHistory
+  let metadata = event.params.metadata.toString()
+
+  colony.metadata = metadata
+
+  let metadataHistory = new ColonyMetadataInstance(
+    event.address.toHex() +
+    '_metadata_' + metadata +
+    '_transaction_' + event.transaction.hash.toHexString() +
+    '_log_' + event.logIndex.toString(),
```

Comment:
I think there's a bug here, since log index, in some cases, returns only `0`

![Screenshot from 2021-01-24 15-03-06](https://user-images.githubusercontent.com/1193222/105631120-6c788180-5e55-11eb-823d-11c72d7f0573.png)